src/subpages/account.tsx

```diff
@@ -138,11 +135,18 @@ const AccountPage: React.FC<AccountPageProps> = ({ location }) => {
                 }}
               />
               <OffsetSwitch
-                olderHandler={olderHandler}
-                newerHandler={newerHandler}
-                disable={{ older: loading, newer: loading || offset < 1 }}
+                olderHandler={mineOlderHandler}
+                newerHandler={mineNewerHandler}
+                disable={{
+                  older: loading || (!!blocks && blocks.length < limit),
```

Comment:
```suggestion
                  older: loading || (!!blocks?.length < limit),
```

Can I update TypeScript version to latest? Additional, I experienced a union type inference bug in this project.

Reply:
Could you give some examples of the bug? I believe it will be oaky If it can resolve the bug by bumping TypeScript's version.

Also, I believe the part you suggested seems not correct. I believe `!!blocks?.length` will be always evaluated as a boolean value. Doesn't it?
```javascript
> const blocks = [1, 2, 3];
> !!blocks?.length
< true
```

Reply:
I checked and It has same issue in latest TS verison. 

| [describe before undefined](https://www.typescriptlang.org/play?ts=4.0.2#code/JYOwLgpgTgZghgYwgAgPIxgZwmAygd2DAQAsAFKAewAdNkBvAKGWRAn2gAk4QATAG2gAuZAAoAlMgC8APmQA3SsF4BuZskr9eXHgOFjJshUtXrewTHABGgkfQ1b9Vypog8Vrdk5eCeyAL5q-oyMoJCwiCjoWDgERKQAspQIANYUNHRMLOaWNhB2DtpQImBQAK4QHmwcxcilFQFBIQiUIJhgaBjYeITEJNJiWZ413HyCUAA06ppFo3qTZhbWglP+ItHdcX3ptMgAPp0xPfEkSak7mIZy9RBAA) | [describe after undefined](https://www.typescriptlang.org/play?ts=4.0.2#code/JYOwLgpgTgZghgYwgAgPIxgZwmAygd2DAQAsAFKAewAdNkBvAKGWRAn2gAk4QATAG2gAuZAAoAlMgC8APmQA3SsF4BuZskr9eXHgOFjJshUtXrewTHABGgkfQ1b9Vypog8Vrdk5eCeyAL5q-oyMoJCwiCjoWDgERKQAspQIANYUNHRMLGwcUNx8glAiAK58EDCgEKYsmtp5uoUlZRVs1cjmljYQdg51ImBQxRAeOfoDQwFBIYwIlCCYYGgY2HiExCTSYlmeufl6UAA06rU6BdBHLB3Wgkf+ItErcevptMgAPksxq-EkSakvmEMcnGECAA) |
| :--: | :--: |
| <img width="532" alt="Screen Shot 2020-08-21 at 11 20 42 AM" src="https://user-images.githubusercontent.com/5278201/90845107-63641f80-e3a0-11ea-91a8-a2d9233a456c.png"> | <img width="475" alt="Screen Shot 2020-08-21 at 11 13 20 AM" src="https://user-images.githubusercontent.com/5278201/90844934-01a3b580-e3a0-11ea-9ae8-b79620bc4534.png"> |

I'll be fix it.

Reply:
```
  Type '{ disable: { older: true; newer: true; }; }' 
is missing the following properties from type 'OffsetSwitchMockProps': newerHandler, olderHandler

59                 <OffsetSwitch disable={{ older: true, newer: true }} />
```

Sadly, It doesn't pass type check :( even prop is undefined like below.

```ts
interface OffsetSwitchMockProps {
  newerHandler: undefined;
  olderHandler: undefined;
  disable: { older: true; newer: true };
}
```